### PR TITLE
Feat/verify endpoint

### PR DESCRIPTION
**Pull request by hiiamtrong**

### CODE REVIEW

File: src/utils/time/time.ts [hunk 1]
```diff
@@ -2,6 +2,10 @@ class TimeUtil {
   static toUnix(date: Date): number {
     return Math.floor(date.getTime() / 1000);
   }
+
+  static isExpired(exp: number): boolean {
```
Comment (gialynguyen) [inline src/utils/time/time.ts:6]: Should use a meaningful name like `expireTimestamp`.

File: src/auth/jwt.strategy.ts [hunk 2]
```diff
@@ -14,10 +19,24 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
...
+      },
+    });
+
+    if (!user) {
+      throw new UnauthorizedException('Invalid credentials');
+    }
+
+    delete user.password;
+
+    return user;
```
Comment (gialynguyen) [inline src/auth/jwt.strategy.ts:40]: Should have a DTO for the `user` response of this method.

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -81,4 +83,19 @@ export class AuthController {
       message: 'success',
     };
   }
+  @UseGuards(JwtAuthGuard)
+  @Get('verify')
+  async verify(@Request() req, @Response() res) {
+    if (!req.user) {
+      throw new UnauthorizedException();
+    }
+    res.set('x-user-id', req.user.id);
```
Comment (gialynguyen) [inline src/auth/auth.controller.ts:92]: add these header field names to the configuration.

**Review by gialynguyen — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma [hunk 4]
```diff
@@ -8,8 +8,8 @@ datasource db {
...
 model UserAuth {
-  id       String @id @default(uuid())
-  email    String @unique @db.VarChar(255)
-  password String @db.VarChar(255)
-
+  id       String   @id @default(uuid())
+  email    String   @unique @db.VarChar(255)
+  password String   @db.VarChar(255)
+  roles    String[]
 }
```
Comment (gialynguyen) [inline prisma/schema.prisma:15]: I think the admin is a special role in our app and the permission mechanism isn't complex. So, we could add an `isAdmin` column.

**Review by gialynguyen — CHANGES_REQUESTED** [review]

File: src/auth/jwt.strategy.ts (hunk 2 above)
Comment (hiiamtrong) [reply src/auth/jwt.strategy.ts:40]: Is that what you mean?
image
So, is there any way to reuse the model UserAuth of PrismaClient

Comment (gialynguyen) [reply src/auth/jwt.strategy.ts:40]: Yes. I think we shouldn't reuse this time to reduce the cross dependency

Comment (hiiamtrong) [reply src/auth/jwt.strategy.ts:41]: Updated

File: src/auth/auth.controller.ts (hunk 3 above)
Comment (hiiamtrong) [reply src/auth/auth.controller.ts]: fixed

File: prisma/schema.prisma (hunk 4 above)
Comment (hiiamtrong) [reply prisma/schema.prisma:16]: updated

File: src/utils/time/time.ts (hunk 1 above)
Comment (hiiamtrong) [reply src/utils/time/time.ts]: fixed

**Review by gialynguyen — APPROVED** [review]

**Pull request merged**